### ISSUE TITLE
Declarer issues

### ISSUE THREAD

**Issue by corysimmons** [posthumans/postcss-ant-grid#1]
- [x] ~~If the expansion of an `ant-*` declaration leaves a ruleset empty, and there are no other custom CSS rules set, it causes all declarations to get an extra level of indentation. Consider using https://github.com/ben-eb/postcss-discard-empty~~ (somehow the indentation problem got fixed with https://github.com/posthumans/postcss-ant-grid/commit/32484f9fad7134cc969c8d535f33a0c2eefce4a4)
- [x] ~~When a ruleset already exists, it makes sense to append to that ruleset rather than create a new ruleset for the same selector. It might be worth finding a plugin that will combine these so we don't leave a big footprint.~~ (this isn't ant's job)
- [x] `declarer()` items are ordered backwards. See if we can get these sorted the other way.
- [x] Comments are repeated for every generated ruleset

Example:
```css
.foo {
  /* comment */
  ant-grid: 1/3 auto;
}
```

```css
.foo {
  /* comment */
}

.foo > * {
  /* comment */
  float: left;
  margin-right: 30px;
}

.foo > *:nth-child(1n) {
  /* comment */
  width: calc(99.99% * 1/3 - (30px - 30px * 1/3));
}

.foo > *:nth-child(2n) {
  /* comment */
  width: calc((99.99% - ((99.99% * 1/3 - (30px - 30px * 1/3))) - (30px * 1)) / 1);
  margin-right: 0;
}
```

- [x] If `ant-*` leaves a ruleset empty, an empty ruleset is leftover. Consider using https://github.com/ben-eb/postcss-discard-empty (or creating a flexible way to add/remove postcss plugins as some of them look pretty snazzy).

Example:
```css
.grid {
  ant-grid: 1/3 auto;
  ant-break: 2;
}
```

```css
.grid {
}
.grid > * {
  float: left;
  margin-right: 20px;
}
.grid > *:nth-child(1n) {
  width: calc(99.99% * 1/3 - (20px - 20px * 1/3));
}
.grid > *:nth-child(2n) {
  width: calc((99.99% - ((99.99% * 1/3 - (20px - 20px * 1/3))) - (20px * 1)) / 1);
  margin-right: 0;
}
```

**Comment by wyze** [comment]
Fixed the 3rd bullet with this commit: 32484f9

**Comment by corysimmons** [comment]
Lovin' it!

**Comment by corysimmons** [comment]
@wyze Actually the declarer stuff is definitely an improvement, but it's still reversing when I split them back up.

Specifically I need to break these apart so I can iterate over one of the `declarer`s like so:

```js
declarer({
  [`${selector} > *`]: {
    float: `left`,
    marginRight: gutter
  }
}, decl)

for (let i = 1; i <= values.length; i++) {
  declarer({
    [`${selector} > *:nth-child(${i}n)`]: {
      width: size
    }
  }, decl)
}

declarer({
  [`${selector} > *:nth-child(${wrap}n)`]: {
    marginRight: 0
  }
}, decl)
```

Some dude in #javascript mentioned using ES6 proxies so I'll look into that.

**Comment by wyze** [comment]
If that doesn't work, you could just structure your code differently to only make one call to `declarer` once you have the full set of selectors built.

```js
const selectors = {
  [`${selector} > *`]: {
    float: `left`,
    marginRight: gutter
  }
}

for (let i = 1; i <= values.length; i++) {
  selectors[`${selector} > *:nth-child(${i}n)`] = {
    width: size
  }
}

selectors[`${selector} > *:nth-child(${wrap}n)`] = {
  marginRight: 0
}

declarer(selectors, decl)
```

**Comment by corysimmons** [comment]
Awesome, works like a charm. =)

**Comment by wyze** [comment]
Empty ruleset was further addressed in this commit: ecb9c17

Which now brings this issue to a close.
